src/components/20-molecules/dropdown/index.js

```diff
@@ -183,15 +187,82 @@ class AXADropdown extends NoShadowDOM {
     const { key, keyCode } = e;
     if (key === 'Escape' || key === 'Esc' || keyCode === 27) {
       e.preventDefault();
-      this.openDropdown(false);
+      return this.openDropdown(false);
+    }
+    return this.handleAutosuggest(key, e);
+  }
+
+  // build one long string of words paired with their indices in `items`,
+  // separating the elements in each pair with suitable control characters
+  // that do not occur in typed input
+  buildAutosuggestDictionary() {
+    this._autosuggestLetters = new Set();
+    const { items, _autosuggestLetters: letters } = this;
+
+    this._autosuggestDictionary =
+      items
+        .filter(item => !item.disabled)
+        .map((item, index) => {
+          // normalize word for purposes of matching
+          const word = item.name.toLowerCase();
+          // construct a set of all letters in word
+          // (used for an early filter over keystrokes)
+          word.split('').forEach(letter => {
+            letters.add(letter.toLowerCase());
+          });
+          return `${INDEX_END}${word}${WORD_END}${index}`;
+        })
+        .join('') + INDEX_END;
+  }
+
+  handleAutosuggest(key) {
+    // ready for autosuggestions, by having focus on the non-native UI?
+    if (document.activeElement !== this.button) {
+      // no, bail out
+      return;
     }
+    // do we have even a chance of matching for the key that was just pressed?
+    const character = key.toLowerCase();
+    if (!this._autosuggestLetters.has(character)) {
+      // no, bail out
+      return;
+    }
+    this._typedSoFar += character;
+    // wait for certain period of keyboard inactivitity...
+    clearTimeout(this._timer);
+    this._timer = setTimeout(() => {
+      // ... before trying to match what was typed sofar against items
+      const { _autosuggestDictionary, _typedSoFar, items } = this;
+      const match = _autosuggestDictionary.indexOf(INDEX_END + _typedSoFar);
+      if (match > -1) {
+        // we have a match, determine the corresponding item ...
+        const matchIndexPos =
+          _autosuggestDictionary.indexOf(WORD_END, match) + 1;
+        const firstSelectedItem = parseInt(
+          _autosuggestDictionary.slice(matchIndexPos),
+          10
+        );
+        // ... and its value
+        const { value } = items[firstSelectedItem];
+        // re-render to visually select matched item
+        this.updateItems(value, 'scrollIntoView');
+      }
+      // next characters start a *new* autosuggestion
+      this._typedSoFar = '';
+    }, AUTOSUGGEST_INACTIVITY_DELAY);
   }
 
   handleDropdownClick(e) {
     e.preventDefault();
     e.stopPropagation();
     // toggle dropdown
-    this.openDropdown(!this.open);
+    const open = !this.open;
```

Comment:
I would rename this to `isOpen` as `open` for me is always a positive value. Just an idea if you see it differently leave it